markets/spot-market/contracts/modules/AtomicOrderModule.sol

```diff
@@ -31,6 +34,11 @@ contract AtomicOrderModule is IAtomicOrderModule {
         SpotMarketFactory.Data storage spotMarketFactory = SpotMarketFactory.load();
         spotMarketFactory.validateMarket(marketId);
 
+        ITokenModule synth = SynthUtil.getToken(marketId);
+        FeatureFlag.ensureAccessToFeature(
+            bytes32(abi.encodePacked(Flags.TRADING_ENABLED, marketId))
```

Comment:
note: while this works fine for progmatic use cases, afaict its not like the human readable text of the flag would be:

```
tradingEnabledSynthId75
```

if the pool ID was 75. we would need to first encode the integer as a string i am pretty sure.

Reply:
You are right, it's hard to decipher which `FeatureUnavailable` is thrown with this setup.

Reply:
I've updated this line to encode the `marketId` to string as well.